src/saveload/de.rs

```diff
@@ -0,0 +1,147 @@
+
+use std::fmt::{self, Display, Formatter};
+use std::marker::PhantomData;
+
+use serde::de::{self, Deserialize, DeserializeSeed, Deserializer, SeqAccess, Visitor};
+
+use {Entities, FetchMut, WriteStorage};
+
+use saveload::{Components, EntityData, Storages};
+use saveload::marker::{Marker, MarkerAllocator};
+
+
+/// Wrapper for `Entity` and tuple of `WriteStorage`s that implements `serde::Deserialize`
+struct DeserializeEntity<'a, 'b: 'a, M: Marker, E, T: Components<M::Identifier, E>> {
+    entities: &'a Entities<'b>,
+    storages: &'a mut <T as Storages<'b>>::WriteStorages,
+    markers: &'a mut WriteStorage<'b, M>,
+    allocator: &'a mut FetchMut<'b, M::Allocator>,
+    pd: PhantomData<(M, E, T)>,
```

Comment:
`M` shouldn't be needed in the `PhantomData` as it's already used in `&'a mut WriteStorage<'b, M>`.

Reply:
Types went serious changes and now it is indeed unnecessary 😄